src/codeenginestudio/huong/Hex.java

```diff
@@ -0,0 +1,51 @@
+package codeenginestudio.huong;
+
+import java.util.Objects;
+
+public class Hex extends Number{
+    Hex(int num){
+        super(num);
+    }
+
+    @Override
+    public String toString() {
+        return "0x" + Integer.toHexString(num).toUpperCase();
+    }
+
+    public static int parse(String input){
+        int hex = Integer.parseInt(input, 16);
```

Comment:
Make "16" to constant variable and return in here, no need create `hex` variable